### ISSUE TITLE
refactor: migrate to OCLIF

### ISSUE THREAD

**Issue by robingenz** [capawesome-team/cli#10]
I really like [citty](https://www.npmjs.com/package/citty), but it lacks many basic functions such as global error handling. [OCLIF](https://oclif.io/) seems to be a grown-up alternative.

**Comment by nicogenz** [comment]
To be honest, another alternative could be to not use any CLI framework at all and instead parse the input ourself. This would allow us to be more flexible instead of depending on other solutions. 
An example can be found here: https://github.com/languine-ai/languine/tree/main/packages/cli

**Comment by robingenz** [comment]
Yes, I also already thought of this. Feel free to create a POC if you would like to work on this. 🙌🏻

**Comment by robingenz** [comment]
Another great library: https://github.com/drizzle-team/brocli

- Easy to use: https://github.com/drizzle-team/drizzle-orm/blob/main/drizzle-kit/src/cli/index.ts
- Supports tests: https://github.com/drizzle-team/drizzle-orm/blob/main/drizzle-kit/tests/cli-generate.test.ts
